From 7ce367bf497465e4a6360bd112ffba025e42541d Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 24 Jul 2018 15:10:57 +0800
Subject: [PATCH 31/65] fix write network conf failed while liveinst

While liveinst, the network conf writing will overwrite copy
"/etc/resolv.conf" to "/mnt/sysimage/etc/resolv.conf". But in
wrlinux, the "/etc/resolv.conf" and "/mnt/sysimage/etc/resolv.conf"
are the same file which is a symlink to /run/resolve.conf.

Test same file before copy.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/network.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pyanaconda/network.py b/pyanaconda/network.py
index d5b0aa7..6eb08d4 100644
--- a/pyanaconda/network.py
+++ b/pyanaconda/network.py
@@ -1214,6 +1214,13 @@ def copyFileToPath(fileName, destPath='', overwrite=False):
         return False
     if not os.path.isdir(os.path.dirname(destfile)):
         util.mkdirChain(os.path.dirname(destfile))
+
+    # Same file test
+    if os.path.exists(fileName) and os.path.exists(destfile) and \
+       os.stat(fileName).st_ino == os.stat(destfile).st_ino:
+        log.info("%s and %s are the same file" % (fileName, destfile))
+        return False
+
     shutil.copy(fileName, destfile)
     return True
 
-- 
2.7.4

